.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+
+  check:
+    needs: []
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Set up Python 3.6"
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: "Install Python dependencies"
+      run: |
+        pip install ".[dev]"
+    - name: "Check code format"
+      run: |
+        make check
```

Comment:
Same comment as previous suggestion:

When does a format-check CI run going to detect issues, that neither the PR creation triggers, nor the master push trigger detected?

Format could be mess up on a particular development branch, and that is OK. It only becomes relevant when it is proposed to be merged to `master`, and the [pre-commit hook](https://github.com/HEPData/hepdata-pyhf-extractor/blob/2fe51eebd92c3103e32b52013a7f53be843e729d/.pre-commit-config.yaml) was not installed (for some reason).

Reply:
> Format could be mess up on a particular development branch, and that is OK. It only becomes relevant when it is proposed to be merged to master, and the pre-commit hook was not installed (for some reason).

I would still advocate for factorization of workflows for different purposes. We used to have everything be in one big GHA workflow file for `pyhf`, but as time has moved on we've factored more and more of it out into separate workflows &mdash; this has worked really well and dropped the entire CI time. While this does introduce additional boilerplate at the top of each workflow file, it is nice to be able to just rerun more specific workflows if there are errors on GitHub's side that cause the workflow to fail. Additionally, if you require all the checks to pass in GitHub branch protections options then you can run all workflows in parallel. Here, this saves almost no times as everything is so small and so fast, but at the same time why not?

Reply:
The main reason to avoid running different jobs in parallel (`check` and `test`), is that one can potentially get more computational expensive than the other one (in this case, `test`). So, you will only want to trigger that job when you are absolutely sure the previous _light-weight_ jobs are successful.

If you move the `check` / `lint` job into another workflow, you are unable to reference it in the `needs: [<job_ID>]` section, and therefore, you will end up spawning `test` jobs without even checking if the style is correct.

This _not-doing-lightweights-checks-first_ methodology could even get more inefficient if a type checker like [Mypy](http://mypy-lang.org/) is introduced. Mypy checks would run faster than a proper test suite run, so it would become another _lightweight_ check that, hopefully, could be referenced in the `needs` section of the `test` job.

Reply:
Okay. It just sounds like you're wanting to optimize the CI differently than I would, which is fine, so I'd just move forward with what you have. 

For `pyhf` we optimize for running all workflows in the shortest amount of wall time for PR. You don't want runs blocked by linting if you care more about making sure if the test suite is going to pass or not.